src/core_tests/workflow_tasks.rs

```diff
@@ -1391,3 +1391,76 @@ async fn buffering_tasks_doesnt_count_toward_outstanding_max() {
         PollWfError::TonicError(_)
     );
 }
+
+#[tokio::test]
+async fn fail_wft_then_recover() {
+    let t = canned_histories::long_sequential_timers(1);
+    let mut mh = MockPollCfg::from_resp_batches(
+        "fake_wf_id",
+        t,
+        [ResponseType::AllHistory, ResponseType::AllHistory],
```

Comment:
Maybe add a comment here to explain that this will cause the same history to be delivered twice?
I had to scroll back up here to understand why WF gets retried.